source/utilities.tex

```diff
@@ -6463,6 +6464,2042 @@
+@\exposid{has_val}@ = false;
+rhs.@\exposid{has_val}@ = true;
+\end{codeblock}
+
+\pnum
+\throws
+Any exception thrown by the expressions in the \Fundescx{Effects}.
+
+\pnum
+\remarks
+The exception specification is equivalent to
+\tcode{is_nothrow_move_constructible_v<E> \&\& is_nothrow_swappable_v<E>}.
+\end{itemdescr}
+
+\indexlibrarymember{swap}{expected<void>}%
+\begin{itemdecl}
+friend constexpr void swap(expected& x, expected& y) noexcept(noexcept(x.swap(y)));
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Equivalent to \tcode{x.swap(y)}.
+\end{itemdescr}
+
+\rSec3[expected.void.obs]{Observers}
+
+\indexlibrarymember{operator bool}{expected<void>}%
+\indexlibrarymember{has_value}{expected<void>}%
+\begin{itemdecl}
+constexpr explicit operator bool() const noexcept;
+constexpr bool has_value() const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+\exposid{has_val}.
+\end{itemdescr}
+
+\indexlibrarymember{operator*}{expected<void>}%
+\begin{itemdecl}
+constexpr void operator*() const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\expects
+\tcode{has_value()} is \tcode{true}.
+\end{itemdescr}
+
+\indexlibrarymember{value}{expected}%
+\begin{itemdecl}
+constexpr void value() const&;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\throws
+\tcode{bad_expected_access(error())} if \tcode{has_value()} is \tcode{false}.
+\end{itemdescr}
+
+\indexlibrarymember{value}{expected<void>}%
+\begin{itemdecl}
+constexpr void value() &&;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\throws
+\tcode{bad_expected_access(std::move(error()))}
+if \tcode{has_value()} is \tcode{false}.
+\end{itemdescr}
+
+\indexlibrarymember{error}{expected<void>}%
+\begin{itemdecl}
+constexpr const E& error() const&;
+constexpr E& error() &;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\expects
+\tcode{has_value()} is \tcode{false}.
+
+\pnum
+\returns
+\exposid{unex}.
+\end{itemdescr}
+
+\indexlibrarymember{error}{expected<void>}%
+\begin{itemdecl}
+constexpr E&& error() &&;
+constexpr const E&& error() const&&;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\expects
+\tcode{has_value()} is \tcode{false}.
+
+\pnum
+\returns
+\tcode{std::move(\exposid{unex})}.
+\end{itemdescr}
+
+\rSec3[expected.void.eq]{Equality operators}
+
+\indexlibrarymember{operator==}{expected<void>}%
+\begin{itemdecl}
+template<class T2, class E2> requires is_void_v<T2>
+  friend constexpr bool operator==(const expected& x, const expected<T2, E2>& y);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\mandates
+The expression \tcode{x.error() == y.error()} is well-formed and
+its result is convertible to \tcode{bool}.
+
+\pnum
+\returns
+If \tcode{x.has_value()} does not equal \tcode{y.has_value()}, \tcode{false};
+otherwise \tcode{x.has_value() || static_cast<bool>(x.error() == y.error())}.
+\end{itemdescr}
+
+\indexlibrarymember{operator==}{expected<void>}%
+\begin{itemdecl}
+template<class E2>
+  constexpr bool operator==(const expected& x, const unexpected<E2>& e);
```

Comment:
```suggestion
  friend constexpr bool operator==(const expected& x, const unexpected<E2>& e);
```
Oops, just realized this from writing my proposal against this...

Reply:
Thanks! Yes, that seems to be missing from the paper. I'll add it.